### DISCUSSION TITLE
Does error.ts need to be a client component?

### DISCUSSION THREAD

**Discussion by codinginflow** [vercel/next.js#49811] [App Router] [answered]
I have an extremely simple `error.tsx` page that doesn't use any client features. Yet, if I don't add `"use client"` to this page, I get an error that says:

```
error Error: Functions cannot be passed directly to Client Components unless you explicitly expose it by marking it with "use server".
  <... parallelRouterKey=... segmentPath=... error={function} errorStyles=... loading=... loadingStyles=... hasLoading=... template=... templateStyles=... notFound=... notFoundStyles=... asNotFound=... 
childProp=... styles=...>
```

This is my `error.tsx`:
```
export default function ErrorPage() {
    return (
        <div>
            <h1>Error 😵</h1>
            <p>Looks like something went wrong. Please refresh the page or contact support.</p>
        </div>
    );
}
```

(Why) does error need to be a client component?

**Comment by icyJoseph** [comment, marked as answer]
Well, according to: https://nextjs.org/docs/app/api-reference/file-conventions/error they `must` be client components, though there's not much rationale for it.

I think it might have to do with the old custom error page, and that it gets passed the actual error - it used to be that the 404 or 500 pages could be pre-rendered, whereas the _error page could be done in demand.

**Reply by codinginflow** [reply]
> Well, according to: https://nextjs.org/docs/app/api-reference/file-conventions/error they `must` be client components, though there's not much rationale for it.

Thank you, I didn't see that. At least I know that my setup is correct!

**Reply by icyJoseph** [reply]
Please don't forget to mark as answered :)
